### PR TITLE
refactor: implement agreed upon changes based on discussion (#266)

### LINKED ISSUE #305: refactor: implement agreed upon changes based on discussion (#266)
## Description
Based on discussion #266, implement the changes agreed upon there as seen in the task list below.

## Tasklist
- create new class `MathUtilities`
- implement proposed change to create `isPowerOfTwo(int)` boolean check inside `MathUtilities`
- create new class `Preconditions`
- implement proposed change to create `requirePowerOfTwo(int)` strict precondition inside `Preconditions`
- implement proposed change to create `isBlank(String)` boolean check inside `StringUtilities`
- implement proposed change to move `StringUtilities#requireNotBlank(String)` to `Preconditions`

### PR DESCRIPTION
**Pull request by ata-nas**
**Description**:

Implement agreed upon changes from discussion #266 

**Related issue(s)**:

Fixes #305 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)
---
- [x] create new class `MathUtilities`
- [x] implement proposed change to create `isPowerOfTwo(int)` boolean check inside `MathUtilities`
- [x] create new class `Preconditions`
- [x] implement proposed change to create `requirePowerOfTwo(int)` strict precondition inside `Preconditions`
- [x] implement proposed change to create `isBlank(String)` boolean check inside `StringUtilities`
- [x] implement proposed change to move `StringUtilities#requireNotBlank(String)` to `Preconditions`
- [x] integrate the new changes for the `MathUtilities` into the respective places

### CODE REVIEW

**Review by georgi-l95 — DISMISSED** [review]
LG

File: server/src/main/java/com/hedera/block/server/notifier/NotifierConfig.java [hunk 1]
```diff
@@ -38,14 +39,8 @@ public record NotifierConfig(@ConfigProperty(defaultValue = "1024") int ringBuff
...
-        if (ringBufferSize <= 0) {
-            throw new IllegalArgumentException("Ring buffer size must be greater than 0");
-        }
-
-        if ((ringBufferSize & (ringBufferSize - 1)) != 0) {
-            throw new IllegalArgumentException("Ring buffer size must be a power of 2");
-        }
-
+        Preconditions.requirePositive(ringBufferSize, "Notifier Ring Buffer Size must be positive!");
+        Preconditions.requirePowerOfTwo(ringBufferSize, "Notifier Ring Buffer Size must be a power of 2!");
```
Comment (mattp-swirldslabs) [inline server/src/main/java/com/hedera/block/server/notifier/NotifierConfig.java:43]: Remove exclamation points

**Review by mattp-swirldslabs — APPROVED** [review]
Looks pretty good. Please remove all the exclamation points from the exception message strings.

File: server/src/main/java/com/hedera/block/server/notifier/NotifierConfig.java (hunk 1 above)
Comment (ata-nas) [reply server/src/main/java/com/hedera/block/server/notifier/NotifierConfig.java]: ✅ Done. Do we have some convention for the error messages? If so, let me know. Thanks!

File: server/src/test/java/com/hedera/block/server/mediator/MediatorConfigTest.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public void testMediatorConfig_happyPath() {
     public void testMediatorConfig_negativeRingBufferSize() {
         IllegalArgumentException exception =
                 assertThrows(IllegalArgumentException.class, () -> new MediatorConfig(-1, ""));
-        assertEquals("Ring buffer size must be greater than 0", exception.getMessage());
+        assertEquals("Mediator Ring Buffer Size must be positive!", exception.getMessage());
```
[the remark this answers is not in the document]
Comment (ata-nas) [reply server/src/test/java/com/hedera/block/server/mediator/MediatorConfigTest.java]: ✅ Done. Do we have some convention for the error messages? If so, let me know. Thanks!

**Review by mattp-swirldslabs — APPROVED** [review]

**Pull request merged**